exp/services/captivecore/internal/server.go

```diff
@@ -0,0 +1,70 @@
+package internal
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/stellar/go/exp/ingest/ledgerbackend"
+	supporthttp "github.com/stellar/go/support/http"
+	"github.com/stellar/go/support/http/httpdecode"
+	supportlog "github.com/stellar/go/support/log"
+)
+
+func serializeResponse(
+	logger *supportlog.Entry,
+	w http.ResponseWriter,
+	r *http.Request,
+	response interface{},
+	err error,
+) {
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(err.Error()))
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		logger.WithContext(r.Context()).WithError(err).Warn("could not serialize response")
```

Comment:
Nit: I guess we can write a header here.
```suggestion
		logger.WithContext(r.Context()).WithError(err).Warn("could not serialize response")
		w.WriteHeader(http.StatusInternalServerError)
```